pkg/plugins/runtime/gateway/generator.go

```diff
@@ -132,39 +145,71 @@ func (g Generator) Generate(ctx xds_context.Context, proxy *core_xds.Proxy) (*co
 			}
 		}
 
-		listener, hosts, err := MakeGatewayListener(ctx.Mesh, gateway, listeners)
+		listener, hosts, err := MakeGatewayListener(ctx, gateway, listeners)
 		if err != nil {
 			return nil, err
 		}
 
 		hosts = RedistributeWildcardRoutes(hosts)
 
+		// Ensure that generators don't get duplicate routes,
+		// which could happen after redistributing wildcards.
```

Comment:
Would it make sense to add a test for this?

Reply:
I'm actually not sure how this is possible, looking at `RedistributeWildcardRoutes`. However, I think it's ok to leave because it's essentially turning a list of resources into a set of resources and a set is definitely the right data structure. I'm putting it into `RedistributeWildcardRoutes` though.